### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by ghost**
Added windows support with also execa library.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,40 +2,118 @@
...
+
+	module.exports.on = () => toggle(true);
+	module.exports.off = () => toggle(false);
+	module.exports.toggle = toggle;
+	module.exports.isOn = () => device().then(isOn);
+	module.exports.restart = () => toggle(false).then(() => toggle(true));
+	module.exports.device = device;
+} else if(process.platform == 'win32') {
+	
+	const isOn = () => {
```
Comment (sindresorhus) [inline index.js:46]: `isOn` should check whether the wireless network card is powered/enabled, not whether it's connected to the internet.

File: index.js [hunk 2]
```diff
@@ -2,40 +2,118 @@
...
+		let isOnline = execa.shell('ping google.com -n 1 -w 1000').then(() => {
+			return new Promise((resolve) => resolve(true));
```
Comment (sindresorhus) [inline index.js:48]: This code is needlessly verbose. You can just return `true` directly.

```js
let isOnline = execa.shell('ping google.com -n 1 -w 1000').then(() => true).catch(() => false);
```

File: index.js [hunk 3]
```diff
@@ -2,40 +2,118 @@
...
+		}).catch(() => {
+			return new Promise((resolve) => resolve(false));
+		});
+
+		return isOnline;
+	}
+
+	const device = () => {
+		let deviceInterface = execa.shell('netsh interface show interface').then((stdout) => {
```
Comment (sindresorhus) [inline index.js:57]: Use `execa()`, not `execa.shell()`. It's faster and safer.

File: index.js [hunk 4]
```diff
@@ -2,40 +2,118 @@
...
-			return toggleDevice(device, turnOn);
+			return new Promise((resolve) => resolve(toggleDevice(turnOn)));
 		}
 
-		return isOn(device).then(isOn => toggleDevice(device, !isOn));
-	});
+		return new Promise((resolve) => resolve(toggleDevice()));
+	};
+
+	const restart = () => {
```
Comment (sindresorhus) [inline index.js:96]: Look at how this is implemented for macOS. It should just reuse the `toggle()` method.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Nice. I think it would make more sense to split the implementations into two files `windows.js` and `macos.js` that is then imported into `index.js` which only does the platform detection.

Also make sure to run `npm test` locally to ensure you follow the code style and that the tests are passing.

File: index.js (hunk 4 above)
Comment (ghost) [reply index.js:96]: I know, but in windows when I run one after the other does not work, the only way I saw is this.

**Comment by Polhp** [comment]
Okey.

File: index.js (hunk 3 above)
Comment (ghost) [reply index.js:57]: The execa only function ``execa()`` in Windows returns an ``(node:3732) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Error: spawn undefined ENOENT``.

**Comment by ghost** [comment]
Okay so I close this pull request and I open another 3 more for the files.

**Comment by sindresorhus** [comment]
Next time, don't open a new pull request, just update the existing one. A new pull requests creates needless noise in the issue tracker and loses the previous discussions.

File: index.js (hunk 4 above)
Comment (sindresorhus) [reply index.js:96]: Then you're doing something wrong. It shouldn't make a difference as `toggle()` executes the exact same thing.

File: index.js (hunk 3 above)
Comment (sindresorhus) [reply index.js:57]: Ensure you're using the `execa` API correctly. The first argument should be the binary and second should be an array of CLI arguments.

**Comment by mastilver** [comment]
> > Next time, don't open a new pull request, just update the existing one. A new pull requests creates needless noise in the issue tracker and loses the previous discussions.
> 
> I already asked you not to open addition pull request. Continue on the previous one instead.

@Polhp simply do the following:
- undelete branch `patch-1`
- git fetch
- git checkout patch-1
- git reset --hard master
(this is do get the content of `master` / #3 into `patch-1`  / #1 )
- git push -f
Voila

**Pull request closed without merging**